### PR TITLE
Create basic CI workflow

### PR DESCRIPTION
**Pull request by velezd**
Workflow for running tests and trying to build documentation

### CODE REVIEW

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+      - name: Build documentation
+        run: |
+          podman run --rm \
+                     -v $GITHUB_WORKSPACE/tclib:/root/tclib:Z \
+                     -w /root/tclib/doc \
+                     tclib make html
+
+      - name: Cleanup
+        if: always()
```
Comment (pholica) [inline .github/workflows/check.yml:45]: I don't know github actions but it sounds to me like `if: always()` could be some kind of default.

File: .github/workflows/check.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    steps:
+      - name: Checkout change
+        uses: actions/checkout@v2
+        with:
+          path: tclib
+
+      - name: Clone test repository
+        run: |
+          git clone --depth 1 ${{ secrets.TEST_REPO }} tests
```
Comment (pholica) [inline .github/workflows/check.yml:24]: Where does the ${{ secrets.TEST_REPO }} come from? Is it some kind of project settings? I'm sorry, I don't know the implementation details of github actions.

**Review by pholica — COMMENTED** [review]
The change looks good, I just don't fully understand how the github actions work.

File: .github/workflows/check.yml (hunk 2 above)
Comment (velezd) [reply .github/workflows/check.yml:24]: It is in Settings of this repository, thanks for reminding me to add it.

File: .github/workflows/check.yml (hunk 1 above)
Comment (pholica) [reply .github/workflows/check.yml:45]: Ooooh, I see, so the default is to have it run if previous steps were successful and the `always()` makes it run even if the previous steps failed like `finally` statement.
Thanks, got it now.

File: .github/workflows/check.yml (hunk 2 above)
Comment (pholica) [reply .github/workflows/check.yml:24]: Ok thanks. Does this also actually mean that tests will have different release cycle than the code? I'm not sure if such thing is desired.

Comment (velezd) [reply .github/workflows/check.yml:24]: Well, that's what we have now. I agree that we should create new tests that can be placed here.

Comment (pholica) [reply .github/workflows/check.yml:24]: I think I know what you mean but I'd like to be sure discussing this via different channel to make sure we mean the same thing.

**Review by pholica — APPROVED** [review]

**Comment by velezd** [comment]
This was merged in #3

**Pull request closed without merging**